src/omv/ports/mimxrt/mimxrt_hal.c

```diff
@@ -87,10 +88,7 @@ void mimxrt_hal_init() {
                          );
     #endif
 
-    // Confgure and enable EDMA
-    edma_config_t edma_config = {0};
-    EDMA_GetDefaultConfig(&edma_config);
-    EDMA_Init(DMA0, &edma_config);
+    dma_init();
```

Comment:
Per our discussion via slack you where okay using the dma poll stored in micropython so that we could share handles with the MicroPython drivers which will use this. I reverted the DMA0/DMAMUX stuff as asked though.

Reply:
Yes only to allocate channels not for initialization, I'd like to rely on as less as possible from the port(s) please keep this in mind.

Reply:
Mmm, looking at the manager it appears I don't have to call dma_init for it to work. It will allow usage of the free pool without calling dma_init(). However, this could lead to bugs in the future not calling dma_init() ourselves as it will try to reinit the EDMA if called by some other code.

Still revert?

Reply:
It's only called from i2s code, but it should be okay to reinit the DMA, in fact I prefer if we revert back to hard-coded channels and just set them as used in `dma_manager.c` (i.e. reserve them).

Reply:
That will require an update of MicroPython too. Should that be in another PR?

Reply:
Reverted the change above. Please let me know if you want me to remove: allocate_dma_channel() from the om_spi_dmi_init() method.

I expect MicroPython will use the dma allocator more in the future. So, we need to reserve the channels in it or use it.

Reply:
> That will require an update of MicroPython too. Should that be in another PR?

Yes please reserve the channels you use in MicroPython, perhaps even find a way to make it configurable with board config files and push it upstream first. For now I guess you can just hard-code them in dma_manager.c

Reply:
Reverted. Will send a separate PR for disabling them in the DMA manager after this PR is merged.